packages/core/src/codewhisperer/util/authUtil.ts

```diff
@@ -402,15 +404,29 @@ export class AuthUtil {
      * auth connections that the Amazon Q extension has cached. We need to remove these
      * as they are irrelevant to the Q extension and can cause issues.
      */
-    public async clearExtraConnections(): Promise<void> {
+    public async clearExtraConnections(source: string): Promise<void> {
         const currentQConn = this.conn
         // Q currently only maintains 1 connection at a time, so we assume everything else is extra.
         // IMPORTANT: In the case Q starts to manage multiple connections, this implementation will need to be updated.
         const allOtherConnections = (await this.auth.listConnections()).filter((c) => c.id !== currentQConn?.id)
         for (const conn of allOtherConnections) {
             getLogger().warn(`forgetting extra amazon q connection: %O`, conn)
-            // in a Dev Env the connection may be used by code catalyst, so we forget instead of fully deleting
-            isInDevEnv() ? await this.auth.forgetConnection(conn) : await this.auth.deleteConnection(conn)
+            await telemetry.auth_modifyConnection.run(async () => {
+                telemetry.record({
+                    connectionState: Auth.instance.getConnectionState(conn) ?? 'undefined',
```

Comment:
I can't tell, but does this include the start url?

Reply:
It's added by `getTelemetryMetadataForConn`